### PR TITLE
chore: Add recommended indexes

### PR DESCRIPTION
**Pull request by sharat87**
This PR adds indexes as recommended by MongoDB cloud.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java [hunk 1]
```diff
@@ -2741,4 +2741,14 @@ private <T extends BaseDomain> T fetchDomainObjectUsingId(String id, MongockTemp
         final T domainObject = mongockTemplate.findOne(query(where(fieldName(path)).is(id)), type);
         return domainObject;
     }
-}
\ No newline at end of file
+
+    @ChangeSet(order = "037", id = "indices-recommended-by-mongodb-cloud", author = "")
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java:2745]: Can we also add the index for newPage collection?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/migrations/DatabaseChangelog2.java:2745]: I was wondering if we should. If that index was the problem, I'm very surprised it didn't show up in the recommended indexes. 🤔 🤷‍♂️

Oh well, no harm I suppose. I'll go ahead and add it here.

**Comment by Rhitottam** [comment]
/perf-test ref=rhito/feat-create-run-meta-before-test-starts-compare-lt-st-for-prev-runs

**Review by mohanarpit — APPROVED** [review]

**Comment by sharat87** [comment]
@Rhitottam, I'll wait for the result of the perf-test? Will it show up as a comment here?

**Comment by Rhitottam** [comment]
> @Rhitottam, I'll wait for the result of the perf-test? Will it show up as a comment here?

Hi I ma just testing a change in our perf-infra code. No need to wait fro the result. 👍🏽

**Pull request merged**